Nexus.Sdk.Token/TokenServerProvider.cs

```diff
@@ -803,6 +808,48 @@ public async Task SubmitOnStellarAsync(IEnumerable<StellarSubmitSignatureRequest
             }
         }
 
+        /// <summary>
+        /// Wait for the completion of a token operation.
+        /// It must be fully processed and either Completed, Failed or Cancelled before returning.
+        /// </summary>
+        /// <param name="code">Unique Nexus identifier of the token operation or transaction hash</param>
+        public async Task<bool> WaitForCompletionAsync(string code, CancellationToken cancellationToken = default)
+        {
+            while (true)
```

Comment:
does it make sense to add an exit condition, i.e. waiting a maximum amount of time / maximum amount of retries?

Reply:
there is - the cancellation token